framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -664,30 +664,37 @@ assert getTopAnnotation(a1) == getTopAnnotation(a2) :
                 "], a2: " + a2 + " [top: " + getTopAnnotation(a2) + "]";
 
         Set<AnnotationMirror> outset = AnnotationUtils.createAnnotationSet();
-        for (AnnotationMirror a1Super : findSmallestTypes(supertypesMap.get(a1))) {
+        for (AnnotationMirror a1Super : supertypesGraph.get(a1)) {
```

Comment:
Thanks for the explanation.   I don't think it needs a separate pull request.  Though in the future if you fix a general bug while working on a feature, a separate pull request as soon as you make the fix would allow others to benefit from the fix faster.  (It would also make me less confused when reviewing a feature pull request :) )